Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Staging/SetpointController.mo

```diff
@@ -0,0 +1,548 @@
+    annotation(Evaluate=true, __cdl(ValueInReference=False), Dialog(tab="Conditionals", group="Staging part load ratio parameters"));
+
+  parameter Real varSpeStaMin(
+    final unit = "1",
+    final min = 0.1,
+    final max = 1)=0.45
+    "Minimum stage up or down part load ratio for variable speed centrifugal stage types"
+    annotation(Evaluate=true, Dialog(enable=anyVsdCen, tab="Conditionals", group="Staging part load ratio parameters"));
+
+  parameter Real varSpeStaMax(
+    final unit = "1",
+    final min = varSpeStaMin,
+    final max = 1)=0.9
+    "Maximum stage up or down part load ratio for variable speed centrifugal stage types"
+    annotation(Evaluate=true, Dialog(enable=anyVsdCen, tab="Conditionals", group="Staging part load ratio parameters"));
+
+  parameter Modelica.SIunits.TemperatureDifference smallTDif = 1
+    "Offset between the chilled water supply temperature and its setpoint for the long condition"
+    annotation(Evaluate=true, Dialog(enable=have_WSE, tab="Conditionals", group="Value comparison parameters"));
+
+  parameter Modelica.SIunits.TemperatureDifference largeTDif = 2
+    "Offset between the chilled water supply temperature and its setpoint for the short condition"
+    annotation(Evaluate=true, Dialog(enable=have_WSE, tab="Conditionals", group="Value comparison parameters"));
+
+  parameter Modelica.SIunits.TemperatureDifference faiSafTDif = 1
+    "Offset between the chilled water supply temperature and its setpoint for the failsafe condition"
+    annotation (Dialog(tab="Conditionals", group="Value comparison parameters"));
+
+  parameter Modelica.SIunits.PressureDifference dpDif = 2 * 6895
+    "Offset between the chilled water pump diferential static pressure and its setpoint"
+    annotation (Dialog(tab="Conditionals", group="Value comparison parameters"));
+
+  parameter Modelica.SIunits.TemperatureDifference TDif = 1
+    "Offset between the chilled water supply temperature and its setpoint for staging down to WSE only"
+    annotation (Dialog(tab="Conditionals", group="Value comparison parameters"));
+
+  parameter Modelica.SIunits.TemperatureDifference TDifHys = 1
+    "Hysteresis deadband for temperature"
+    annotation (Dialog(tab="Conditionals", group="Value comparison parameters"));
+
+  parameter Modelica.SIunits.PressureDifference faiSafDpDif = 2 * 6895
+    "Offset between the chilled water differential pressure and its setpoint"
+    annotation (Dialog(tab="Conditionals", group="Value comparison parameters"));
+
+  parameter Modelica.SIunits.PressureDifference dpDifHys = 0.5 * 6895
+    "Pressure difference hysteresis deadband"
+    annotation (Dialog(tab="Conditionals", group="Value comparison parameters"));
+
+  parameter Real effConSigDif = 0.05
+    "Signal hysteresis deadband"
+    annotation (Dialog(tab="Conditionals", group="Value comparison parameters"));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uWseSta if have_WSE
```

Comment:
make sure the order of the interface connector is the same showing in the diagram layer. For example in this case, `TChiWatSupSet` should be the first instance and `TChiWatSup` is the second one, and so on.

Reply:
Why is this relevant?